awx/ui/src/screens/WorkflowApproval/shared/WorkflowApprovalControls.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { t } from '@lingui/macro';
+import {
+  Dropdown as PFDropdown,
+  DropdownItem,
+  Tooltip,
+  DropdownToggle,
+  DropdownToggleAction,
+} from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { useKebabifiedMenu } from 'contexts/Kebabified';
+import styled from 'styled-components';
+
+const Dropdown = styled(PFDropdown)`
+  --pf-c-dropdown__toggle--disabled--BackgroundColor: var(
+    --pf-global--disabled-color--200
+  );
+
+  &&& {
+    button {
+      color: var(--pf-c-dropdown__toggle--m-primary--Color);
+    }
+    div.pf-m-disabled > button {
+      color: var(--pf-global--disabled-color--100);
+    }
+  }
+`;
+
+function isPending(item) {
+  return item.status === 'pending';
+}
+function isComplete(item) {
+  return item.status !== 'pending';
+}
+
+function WorkflowApprovalControls({
+  selected,
+  onHandleDeny,
+  onHandleCancel,
+  onApprove,
+  onToggleToolbarKebab,
+  isKebabOpen,
+}) {
+  const { isKebabified } = useKebabifiedMenu();
+
+  const hasSelectedItems = selected.length > 0;
+  const isApproveDenyOrCancelDisabled =
+    !hasSelectedItems ||
+    !selected.every(
+      (item) => item.status === 'pending' && item.can_approve_or_deny
+    );
+
+  const renderTooltip = (action) => {
+    const completedItems = selected
+      .filter(isComplete)
+      .map((item) => item.name)
+      .join(', ');
+    if (!hasSelectedItems) {
+      return t`Select items to approve, deny, or cancel`;
+    }
+    if (!selected.some(isPending)) {
+      return t`Cannot approve, cancel or deny completed workflow approvals`;
+    }
+    if (selected.some(isPending) && selected.some(isComplete)) {
```

Comment:
We could move the definition of `completedItems` closer to where is used. The earlier returns may not even need to calculate that variable.

```suggestion
    if (selected.some(isPending) && selected.some(isComplete)) {
    const completedItems = selected
      .filter(isComplete)
      .map((item) => item.name)
      .join(', ');
```